### PR TITLE
titan: Introduce titan zstd dict compression

### PR DESCRIPTION
**Pull request by Connor1996**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/5743

What's Changed:

```commit-message
Introduce titan zstd dict compression
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: etc/config-template.toml [hunk 1]
```diff
@@ -927,6 +927,12 @@
 ## default: lz4
 # blob-file-compression = "lz4"
 
+## Set blob file zstd dictionary compression, default(0) will use zstd compression
+## It is recommend to set dictionary size to 4k, 16k, and sample data to train 
+## dictionary is of size 100X dictionary size.
+## It has no effect when `blob-file-compression` is not `zstd`.
+# zstd-dict-size = 0
```
Comment (tonyxuqqi) [inline etc/config-template.toml:934]: Why the default value is 0 but the recommended size is 4K or 16K.

Comment (Connor1996) [reply etc/config-template.toml:934]: It means when enabling the dictionary compression, recommend to set size to 4k/16k

**Review by v01dstar — APPROVED** [review]
LGTM

**Review by tonyxuqqi — APPROVED** [review]

File: etc/config-template.toml [hunk 2]
```diff
@@ -927,6 +927,12 @@
 ## default: lz4
 # blob-file-compression = "lz4"
 
+## Set blob file zstd dictionary compression, default(0) will use zstd compression
+## It is recommend to set dictionary size to 4k, 16k, and sample data to train 
+## dictionary is of size 100X dictionary size.
+## It has no effect when `blob-file-compression` is not `zstd`.
```
Comment (LykxSassinator) [inline etc/config-template.toml:933]:
```suggestion
## It has no effect when `blob-file-compression` is not `zstd`.
## default: 0
```

File: etc/config-template.toml [hunk 3]
```diff
@@ -927,6 +927,12 @@
 ## default: lz4
 # blob-file-compression = "lz4"
 
+## Set blob file zstd dictionary compression, default(0) will use zstd compression
+## It is recommend to set dictionary size to 4k, 16k, and sample data to train 
+## dictionary is of size 100X dictionary size.
```
Comment (LykxSassinator) [inline etc/config-template.toml:932]:
```suggestion
## Set blob file zstd dictionary compression, default(0) will use zstd compression.
## It is recommended to set the dictionary size to values such as 4k or 16k. Additionally,
## the sample data size to train dictionary is of size 100X dictionary size by default.
```

**Review by LykxSassinator — APPROVED** [review]

**Pull request merged**